July-16-2015/Solutions/Problem3Solution.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+Class Node {
+    public $value;
+    public $next;
+}
+
+Class Stack {
+    private $head;
+    private $min;
+
+    public function __construct() {
+        $this->head = null;
+        $this->min = null;
+    }
+
+    public function push($value) {
+        $node = new Node();
+        $node->next = null;
+        $node->value = $value;
+
+        if ($this->head === null) {
+            $this->head = $node;
+            $this->min = $node;
+        } else {
+            $node->next = $head;
+            $this->head = $node;
+
+            if ($node->value < $this->min->value) { // don't remember quite how $this-> syntax works with multiple levels
+                $this->min = $node;
+            }
+        }
+
+    }
+
+    public function pop() {
```

Comment:
think all thats left was just moving head to point back 1, and then iterating over all the values in the list to properly set the min (this was the part we were unhappy with) and then returning the node head used to point to